### PR TITLE
feat: development of seedlot review step 5 interim storage

### LINKED ISSUE #833: Development of Seedlot Review step 5 - Interim Storage
Describe the task:

Create FE of Seedlot Review, Step 5 (interim storage) and integrate endpoints as you go along.

Figma design here: https://www.figma.com/file/ROgZJQezb1DDG4AvEgjtpw/Registration-Process----Reviewer?type=design&node-id=2744-284935&mode=design&t=XxxbormmPMPHmENb-4

Image

Acceptance Criteria
- [ ] Developed as per Figma
- [ ] relevant architecture documentation updated
- [ ] tested and merged
- [ ] Develop floater save button place holder - design to be tested by TSC staff

### PR DESCRIPTION
**Pull request by mgaseta**
# Description
Closes #833 

#### New
- Created new Interim read and edit components
- Added them to the seedlot review view

#### Changed
- Added new `isReview` prop on the ownership step
- Changed `CollectionStep` props type name

#### Removed
- Unused types associated with interim step

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-45-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1145-backend.apps.silver.devops.gov.bc.ca/actuator/health)
- [Oracle-API](https://nr-spar-1145-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by RMCampos** [comment]
Waiting for https://github.com/bcgov/nr-spar/pull/1177

File: frontend/src/components/SeedlotReviewSteps/Interim/Read/index.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      <Row>
+        <Column className="sub-section-title-col">
+          Interim agency
+        </Column>
+      </Row>
+      <Row>
+        <Column className="info-col" sm={4} md={4} lg={4}>
+          <ReadOnlyInput
+            id="interim-agency-name"
+            label="Interim agency acronym"
```
Comment (RMCampos) [inline frontend/src/components/SeedlotReviewSteps/Interim/Read/index.tsx:56]: question: when reviewing it has the "acronym" word, but when editing there's only "intering agency", is this how it should be?

**Review by RMCampos — APPROVED** [review]
Great job! 🥇  Only a question!

**Review by RMCampos — APPROVED** [review]
LGTM 🚀

**Review by craigyu — COMMENTED** [review]
image
image

The interim's agency selection does not change if the collector's agency is changed. I suggest that we disable the checkboxes for agency selection on the seedlot review form, let's discuss this with @mmarsoleta on Monday the 27th.

**Review by craigyu — APPROVED** [review]

**Pull request merged**